File: src/MaksIT.WebUI/src/models/identity/user/UserResponse.ts
import { ResponseModelBase } from '../../ResponseModelBase'
import { ScopeEntityType, ScopePermission } from '../../engine/scopeEnums'

/** Matches <c>MaksIT.CertsUI.Models.Identity.User.UserResponse</c> (camelCase JSON). */
export interface UserResponse extends ResponseModelBase {
  id: string
  username: string
  email?: string | null
  mobileNumber?: string | null
  isActive: boolean

  twoFactorEnabled?: boolean
  twoFactorRecoveryCodes?: string[]
  qrCodeUrl?: string
  recoveryCodesLeft?: number

  isGlobalAdmin?: boolean
  entityScopes?: UserEntityScopeResponse[]
}

export interface UserEntityScopeResponse {
  id: string
  entityId: string
  entityType: ScopeEntityType
  scope: ScopePermission
}

File: src/MaksIT.Models/LetsEncryptServer/Identity/User/UserResponse.cs
using MaksIT.Core.Abstractions.Webapi;


namespace MaksIT.Models.LetsEncryptServer.Identity.User;

public class UserResponse : ResponseModelBase {

  #region Master data Properties

  public Guid Id { get; set; }
  public string? Username { get; set; }
  public bool IsActive { get; set; }
  public DateTime? LastLogin { get; set; }

  #endregion

  #region Two-factor authentication properties

  public bool TwoFactorEnabled { get; set; }
  public List<string>? TwoFactorRecoveryCodes { get; set; }
  public string? QrCodeUrl { get; set; }
  public int? RecoveryCodesLeft { get; set; }

  #endregion
}
